### PR TITLE
wait_fixed() to wait_exponential()

### PR DESCRIPTION
**Pull request by IreneTenison**
Use exponential backoff when retrying with tenacity #1257

### CODE REVIEW

File: scripts/commit_classifier.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def clone_git_repo(self, repo_url, repo_dir, rev="master"):
                 lambda: subprocess.run(
                     ["git", "clone", "--quiet", repo_url, repo_dir], check=True
                 ),
-                wait=tenacity.wait_fixed(30),
+                wait=tenacity.wait_exponential(),
                 stop=tenacity.stop_after_attempt(5),
             )()
```
Comment (marco-c) [inline scripts/commit_classifier.py]: What is the total time we would wait with the new code vs the old code?

Comment (IreneTenison) [reply scripts/commit_classifier.py]: wait_fixed(30) would be waiting for 30 seconds between retries.
wait_exponential() would wait for 2^x * 1 second between each retry starting with 0 seconds, then up to 1073741823 seconds. 
should the max time be limited to 30 seconds?

Comment (marco-c) [reply scripts/commit_classifier.py]: So with wait_fixed we were waiting 30 seconds * 5 times = 150 seconds. We should ensure we have a similar total with wait_exponential. If it is 2^x * 1 up to x == 4 (if it starts from 0), it is 1 second, 2 seconds, 4 seconds, 8 seconds, 16 seconds -> in total 31 seconds.
So we should:
- start with waiting more than just 1 second (probably with 4);
- stop after we have waited in total more than just 31 seconds.

Comment (IreneTenison) [reply scripts/commit_classifier.py]: wait_exponential(multiplier = 1, min = 4, max = 31) would wait for 4 seconds, 8 seconds, 16 seconds,  a total of 28 seconds (less than 30s). If max = 32, the total wait time would be 60 seconds, which would be too high. Is it okay if the total wait time is 28 seconds?

Comment (marco-c) [reply scripts/commit_classifier.py]: Before your PR we were waiting 150 seconds in total, so 60 seconds is not too high, it's too low :)

Comment (IreneTenison) [reply scripts/commit_classifier.py]: wait_exponential(multiplier = 1, min = 4, max = 150) would wait for a maximum of 150 seconds. Is this fine? I have done a PR with this change as well.

Comment (marco-c) [reply scripts/commit_classifier.py]: This doesn't work as you expect, try it in a standalone script or read the documentation (https://tenacity.readthedocs.io/en/latest/#waiting-before-retrying). It is waiting forever, never stopping.

Comment (marco-c) [reply scripts/commit_classifier.py]: E.g. you can try with this script:
```Python
import tenacity

@tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, min=4, max=150))
def wait_test():
    print("qua")
    raise Exception

wait_test()
```

**Comment by marco-c** [comment]
@IreneTenison are you still planning on working on this PR?

**Comment by marco-c** [comment]
Closing for lack of response.

**Pull request closed without merging**